File: src/main/java/com/braggbnb108/service/HostReviewService.java
package com.braggbnb108.service;

import java.util.List;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.domain.Specification;
import org.springframework.http.ResponseEntity;

import com.braggbnb108.domain.HostReview;
import com.braggbnb108.dto.HostReviewDTO;
import com.braggbnb108.dto.HostReviewSearchDTO;
import com.braggbnb108.dto.HostReviewPageDTO;
import com.braggbnb108.dto.HostReviewConvertCriteriaDTO;
import com.braggbnb108.service.GenericService;
import com.braggbnb108.dto.common.RequestDTO;
import com.braggbnb108.dto.common.ResultDTO;
import java.util.List;
import java.util.Optional;





public interface HostReviewService extends GenericService<HostReview, Integer> {

	List<HostReview> findAll();

	ResultDTO addHostReview(HostReviewDTO hostReviewDTO, RequestDTO requestDTO);

	ResultDTO updateHostReview(HostReviewDTO hostReviewDTO, RequestDTO requestDTO);

    Page<HostReview> getAllHostReviews(Pageable pageable);

    Page<HostReview> getAllHostReviews(Specification<HostReview> spec, Pageable pageable);

	ResponseEntity<HostReviewPageDTO> getHostReviews(HostReviewSearchDTO hostReviewSearchDTO);
	
	List<HostReviewDTO> convertHostReviewsToHostReviewDTOs(List<HostReview> hostReviews, HostReviewConvertCriteriaDTO convertCriteria);

	HostReviewDTO getHostReviewDTOById(Integer hostReviewId);







}

File: braggbnb108_ui/src/services/HostReviewService.js
import http from "../http-common"; 

class HostReviewService {
  getAllHostReviews(searchDTO) {
    console.log(searchDTO)
    return this.getRequest(`/hostReview/hostReviews`, searchDTO);
  }

  get(hostReviewId) {
    return this.getRequest(`/hostReview/${hostReviewId}`, null);
  }

  findByField(matchData) {
    return this.getRequest(`/hostReview?field=${matchData}`, null);
  }

  addHostReview(data) {
    return http.post("/hostReview/addHostReview", data);
  }

  update(data) {
  	return http.post("/hostReview/updateHostReview", data);
  }
  
  uploadImage(data,hostReviewId) {
  	return http.postForm("/hostReview/uploadImage/"+hostReviewId, data);
  }




	postRequest(url, data) {
		return http.post(url, data);
      };

	getRequest(url, params) {
        return http.get(url, {
        	params: params
        });
    };

}

export default new HostReviewService();
